ejb3/src/main/java/org/jboss/as/ejb3/iiop/EjbCorbaServant.java

```diff
@@ -314,21 +324,59 @@ public OutputStream _invoke(final String opName, final InputStream in, final Res
                                 retVal = false;
                             }
                         } else {
-
-                            if (sc != null) {
-                                setSecurityContextOnAssociation(sc);
+                            if (this.securityDomain != null) {
+                                // an elytron security domain is available: authenticate and authorize the client before
+                                // invoking the component.
+                                ServerAuthenticationContext context = this.securityDomain.createNewAuthenticationContext();
+                                if (credential == sasCurrent) {
+                                    // TODO change this to use the SASCurrent as the credential when we have a realm that can act on it.
```

Comment:
Good point about the IdentityToken and it was something I was also considering. However I don't the chain will always be there. Of course, we can add the code to handle that in the event of the chain being there, but what should we do if all we get is an IdentityToken with a principal, no credentials set?